src/core/translate.ml

```diff
@@ -53,11 +53,15 @@ let var_of_arg ~driver arg : Translated.ocaml_var =
 let type_ ~driver ~ghost (td : Tast.type_declaration) =
   let name = td.td_ts.ts_ident.id_str in
   let loc = td.td_loc in
-  let mutable_ = false (* XXX FIXME look at the ocaml type *) in
+  let mutable_ = Mutability.type_declaration ~driver td in
   let type_ = type_ ~name ~loc ~mutable_ ~ghost in
   let process ~type_ (spec : Tast.type_spec) =
     let term_printer = Fmt.str "%a" Tterm.print_term in
-    let mutable_ = type_.mutable_ || spec.ty_ephemeral in
+    let mutable_ =
+      max
```

Comment:
I think using the polymorphic `max` is a bit fragile here. Perhaps you can define an explicit one in `Mutability`?